File: 練習問題/1-5/問題4：リスト内の最大値を見つける.py
def find_max(arr: list[int]) -> int:
    maxValue: int = arr[0]

    for value in arr[1:]:
        if value > maxValue:
            maxValue = value
    return maxValue


def main() -> None:
    searchArray: list[int] = [20, 3, 6, 8, -10, 6]

    print(find_max(searchArray))


main()

File: 練習問題/1-5/問題4：リスト内の最大値を見つける.ts
const findMax = (arr: Array<number>) => {
    let maxValue = arr[0];

    for (let value: number = 1; value < arr.length; value++) {
        if (arr[value] > maxValue) maxValue = arr[value];
    }
    return maxValue;
};

const main = () => {
    const list: Array<number> = [6, 9, 111, -65, 4, 687, 0, 12];

    return findMax(list);
};

console.log(main());
